frontend/src/views/files/OnlyOfficeEditor.vue

```diff
@@ -0,0 +1,154 @@
+<template>
+  <div id="editor-container">
+    <header-bar>
+      <action icon="close" :label="$t('buttons.close')" @action="close()" />
+      <title>{{ req.name }}</title>
+    </header-bar>
+
+    <breadcrumbs base="/files" noLink />
+
+    <div id="editor"></div>
+  </div>
+</template>
+
+<style scoped>
+#editor-container {
+  height: 100vh;
+  width: 100vw;
+}
+</style>
+
+<script>
+import { mapState } from "vuex";
+import url from "@/utils/url";
+import { baseURL, onlyOffice } from "@/utils/constants";
+
+import HeaderBar from "@/components/header/HeaderBar";
+import Action from "@/components/header/Action";
+import Breadcrumbs from "@/components/Breadcrumbs";
+
+export default {
+  name: "onlyofficeeditor",
+  components: {
+    HeaderBar,
+    Action,
+    Breadcrumbs,
+  },
+  data: function () {
+    return {};
+  },
+  computed: {
+    ...mapState(["req", "user", "jwt"]),
+    breadcrumbs() {
+      let parts = this.$route.path.split("/");
+
+      if (parts[0] === "") {
+        parts.shift();
+      }
+
+      if (parts[parts.length - 1] === "") {
+        parts.pop();
+      }
+
+      let breadcrumbs = [];
+
+      for (let i = 0; i < parts.length; i++) {
+        breadcrumbs.push({ name: decodeURIComponent(parts[i]) });
+      }
+
+      breadcrumbs.shift();
+
+      if (breadcrumbs.length > 3) {
+        while (breadcrumbs.length !== 4) {
+          breadcrumbs.shift();
+        }
+
+        breadcrumbs[0].name = "...";
+      }
+
+      return breadcrumbs;
+    },
+  },
+  created() {
+    window.addEventListener("keydown", this.keyEvent);
+  },
+  beforeDestroy() {
+    window.removeEventListener("keydown", this.keyEvent);
+    this.editor.destroyEditor();
+  },
+  mounted: function () {
+    let onlyofficeScript = document.createElement("script");
+    onlyofficeScript.setAttribute(
+      "src",
+      `${onlyOffice}/web-apps/apps/api/documents/api.js`
+    );
+    document.head.appendChild(onlyofficeScript);
+
+    onlyofficeScript.onload = () => {
+      let fileUrl = `${window.location.protocol}//${window.location.host}${baseURL}/api/raw${url.encodePath(
+        this.req.path
+      )}?auth=${this.jwt}`;
+
+      let key = Date.parse(this.req.modified).toString() + url.encodePath(this.req.path);
+      key = key.replaceAll(/[-_.!~[\]*'()/,;:\-%+.]/g, "");
```

Comment:
might want a comment explaining this regex, it's pretty intense

Reply:
To me it looks like he's removing the following chars form the key `-`, `_`, `.`, `!`, `~`, `[`, `]`, `*`, `'`, `(`, `)`, `/`, `,`, `;`, `:`, `-`, `%`, `+`, `.`,